WorkloadManagementSystem/Client/WMSClient.py

```diff
@@ -138,13 +143,49 @@ def submitJob( self, jdl, jobDescriptionObject = None ):
       return result
 
     # Submit the job now and get the new job ID
+    bulkTransaction = self.operationsHelper.getValue( 'JobScheduling/BulkSubmissionTransaction', False )
+    if bulkTransaction:
+      result = getParameterVectorLength( classAdJob )
+      if not result['OK']:
+        return result
+      nJobs = result['Value']
+      parametricJob = nJobs > 0
+
     if not self.jobManager:
       self.jobManager = RPCClient( 'WorkloadManagement/JobManager',
                                    useCertificates = self.useCertificates,
                                    timeout = self.timeout )
-    result = self.jobManager.submitJob( classAdJob.asJDL() )
-    if 'requireProxyUpload' in result and result['requireProxyUpload']:
+    if bulkTransaction and parametricJob:
+      gLogger.debug( 'Applying transactional job submission' )
+      resultSubmit = self.jobManager.submitJob( classAdJob.asJDL(), True )
+      if resultSubmit['OK'] and resultSubmit.get( 'requireBulkSubmissionConfirmation' ):
```

Comment:
what is this requireBulkSubmissionConfirmation ?
The code is clear in general, but maybe a few comments for all this block would help getting the logic :-)